File: Java/Desktop/ReadyToExam/src/NewYearGift/Model/Entity/Abstract/Candy.java
package NewYearGift.Model.Entity.Abstract;

public class Candy {

    private double weight;
    private String creator;
    private double price;


    public Candy(double weight, String creator, double price) {
        this.weight = weight;
        this.creator = creator;
        this.price = price;

    }


    public double getPrice() {
        return price;
    }

    public void setPrice(double price) {
        this.price = price;
    }

    public double getWeight() {
        return weight;
    }

    public void setWeight(double weight) {
        this.weight = weight;
    }

    public String getCreator() {
        return creator;
    }

    public void setCreator(String creator) {
        this.creator = creator;
    }

    @Override
    public boolean equals(Object object) {
        if (object == null || this.getClass() != object.getClass()){
            return false;
        }else{
            return true;
        }
    }

    @Override
    public String toString() {
        return new StringBuilder().append("Parameters:\nWeight: ").append(weight).append("\nCreator: ").append(creator).append("\nPrice: ").append(price).append("\n").toString();
    }
}

File: Csharp/ReadyToExam/ReadyToExam/NewYearGift/Model/Entity/Abstracts/Candy.cs
using System;
using System.Text;

namespace ReadyToExam.NewYearGift.Model.Abstracts
{
    public class Candy
    {
        
        private double _weight;
        private string _creator;
        private double _price;

        public Candy(double weight, string creator, double price)
        {
            this._weight = weight;
            this._creator = creator;
            this._price = price;
        }

        public double Weight
        {
            get => _weight;
            set => _weight = value;
        }

        public string Creator
        {
            get => _creator;
            set => _creator = value;
        }

        public double Price
        {
            get => _price;
            set => _price = value;
        }

        public override bool Equals(object obj)
        {
            if (obj == null || this.GetType().Name == obj.GetType().Name)
            {
                return true;
            }
            return false;
        }

        public override string ToString()
        {
            return new StringBuilder().Append("Params:\nWeight = " + _weight + "\nCreator = " + _creator + "\nPrice = " + _price).ToString();
        }
        
    }
}